lesson10/script.js

```diff
@@ -0,0 +1,55 @@
+function showLoadingImg(){
+    const loadingImg = document.createElement('img');
+    loadingImg.src = "loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+function getDataAfter3Seconds(){
+    return new Promise((resolve,reject) => {
+        const categriesData = [
+            {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+            {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+        ];
+        setTimeout(() => {
+            if(categriesData){
+                resolve(categriesData);
+            }else{
+                reject(new Error('データの取得ができませんでした'));
+            }
+        },3000);
```

Comment:
I thought that I couldn't catch the error without the condition, so I'm doing conditional branching.

If there is categorical data, it will be resolved.
If there is no category data, it will be rejected.

Is there any other way?

Reply:
Thank you for your reply!
I was concerned that CATCH was not working if categoriesData did not exist.  If that is the reason, I thought catch should work. :astonished: